.golangci.yml

```diff
@@ -0,0 +1,47 @@
+run:
+  timeout: 10m
+  issues-exit-code: 1
+  tests: true
+  skip-dirs-use-default: true
+  modules-download-mode: readonly
+  allow-parallel-runners: false
+
+linters:
+  fast: false
+  enable:
+  - gofmt
+  - goimports
+  - revive
+  - govet
+  - misspell
+  - exportloopref
+  disable:
+  - scopelint
+  disable-all: false
+  presets:
+  - bugs
+  - unused
+
+# all available settings of specific linters
+linters-settings:
+  gofmt:
+    # simplify code: gofmt with `-s` option, true by default
+    simplify: true
+  golint:
+    min-confidence: 0.9
+  govet:
+    # report about shadowed variables
+    check-shadowing: true
+  misspell:
+    locale: US
+
+issues:
+  exclude-rules:
+    # Exclude some linters from running on tests files.
+    - path: _test\.go
+      linters:
+        - gocyclo
+        - errcheck
+        - dupl
+  exclude:
+    - Using the variable on range scope `tc` in function literal
```

Comment:
I'm curious, did you copy this `.golangci.yml` from somewhere, or did you hand-select these settings?

Reply:
No, I copied it from [here](https://github.com/kubernetes-sigs/gateway-api/blob/main/.golangci.yml), but I removed one of the rules because I didn't think it was working here, so I removed it.

If this is necessary, I can add it back.

https://github.com/kubernetes-sigs/gateway-api/blob/6445cc46d294bc61b44a37315ed51bf1597bcbbd/.golangci.yml#L30

https://github.com/kubernetes-sigs/gateway-api/blob/6445cc46d294bc61b44a37315ed51bf1597bcbbd/.golangci.yml#L39

Reply:
The reason why I removed `ignore-words` is that my vscode has some lint rules, and `ignore-words` does not take effect here, and it will cause the editor to report an error. 

Of course, if this rule has other effects, it can be restored

Reply:
No I think this is fine, thank you for the explanation :+1: